music21/musicxml/m21ToXml.py

```diff
@@ -3691,8 +3691,9 @@ def setNoteInstrument(self,
             return
 
         searchingObject = chordParent if chordParent else n
-        closest_inst = searchingObject.getInstrument()
-        if closest_inst is None:
+        closest_inst = searchingObject.getInstrument(returnDefault=True)
+        if closest_inst is None:  # pragma: no cover
+            # unreachable; purely for extra safety
```

Comment:
Should we add a `returnDefault=` argument to match `Stream.getInstrument()`. That's how I ended up misinterpreting this.